src/System.IO.FileSystem/src/System/IO/PathHelpers.Unix.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO
+{
+    internal static partial class PathHelpers
+    {
+        /// <summary>
+        /// Returns fully normalized path.
```

Comment:
The caller is making an assumption that if the returned value is the exact same length as the path passed in, then it is identical. ie normalization always changes the lenght. If that's true, it should be documented here.

Reply:
this bit :` return pathSpan.Length == path.Length ? FileSystem.FileExists(path) : FileSystem.FileExists(new string(pathSpan));`